### ISSUE TITLE
Failure to remove namespaces using base template processor

### ISSUE THREAD

**Issue by vinny-sabatini** [KohlsTechnology/eunomia#278]
**What version of eunomia are you using?**

eunomia version: v0.1.1

**Does this issue reproduce with the latest release?**

Yes

**What operating system and processor architecture are you using (`kubectl version`)?**

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
OCP 3.11
</pre></details>

**What did you do?**

Created a GitOpsConfig using the base template processor (shown below).
In the git repository for that GitOpsConfig, I added a namespace definition and allowed it to be provisioned by Eunomia.
After the namespace was provisioned, I removed the namespace definition from the repository.
```
---
apiVersion: eunomia.kohls.io/v1alpha1
kind: GitOpsConfig
metadata:
  name: namespace-poc
spec:
  templateSource:
    uri: https://github.example.com/operations/openshift-namespaces
    ref: bug-tester
    contextDir: mycluster/namespaces
    secretRef: eunomia-runner-github
  triggers:
  - type: Webhook
  serviceAccountRef: eunomia-runner
  templateProcessorImage: docker-registry.default.svc:5000/default/eunomia-base:latest
  resourceHandlingMode: Apply
  resourceDeletionMode: Delete
```

**What did you expect to see?**
I expected the next Eunomia job to re-apply all of the resources defined in the repository, and then the namespace that was removed from the repository to be marked for deletion.

**What did you see instead?**
The job re-applied all resources defined in the repository, but the namespace that was deleted from git was not marked for deletion. At the end of the log, there was an error stating `error: the server doesn't have a resource type "Status"`.
Ref [remove-namespace.log](https://github.com/KohlsTechnology/eunomia/files/4136341/remove-namespace.log)

**Additional Context**
Based on the error message, I believe this is tightly coupled with #277

**Comment by akavel** [comment]
My initial hypothesis would be that this and #277 might all be different manifestations of the same root cause as in #273.

**Comment by seanmalloy** [comment]
Fixed by #302
